src/core/execute.c

```diff
@@ -5502,6 +5508,18 @@ void exec_context_dump(const ExecContext *c, FILE* f, const char *prefix) {
                 fputs("\n", f);
         }
 
+        if (!strv_isempty(c->exec_paths)) {
+                fprintf(f, "%sExecPaths:", prefix);
+                strv_fprintf(f, c->exec_paths);
+                fputs("\n", f);
+        }
+
+        if (!strv_isempty(c->no_exec_paths)) {
+                fprintf(f, "%sNoExecPaths:", prefix);
+                strv_fprintf(f, c->no_exec_paths);
+                fputs("\n", f);
+        }
```

Comment:
hmm, maybe add a helper call for this instead of repeating this even more... maybe:

```c
static void exec_context_dump_paths(const char *field, char **list) {
…
```

Reply:
Maybe the structure in load-fragment-gperf could also refer to dump functions. For example:
```
$1.ReadWriteDirectories,                 config_parse_namespace_path_strv,            0,                                  offsetof($1, exec_context.read_write_paths)
```
would become
```
$1.ReadWriteDirectories,                 config_parse_namespace_path_strv,            dump_path_strv, 0,                                  offsetof($1, exec_context.read_write_paths)
```
where dump_path_strv() would be something like
```
void dump_path_strv(const ExecContext *c, FILE* f, const char *prefix, size_t field_offset, const char *name) {
        const char *strv = (const char**)(size_t)c + field_offset);
        if (!strv_isempty(strv) {
                fprintf(f, "%s%s:", prefix, name);
                strv_fprintf(f, strv);
                fputs("\n", f);
        }
}
```

Then exec_context_dump() would iterate the structure and call the dump functions for each field. Perhaps it should calculate the pointer to the item instead of each function doing it.

Reply:
that doesn't really work, since this doesn't map 1:1...

Reply:
How about a set of macros then?
```
#define STRV_PRINTF(context, file, prefix, field, name) \
        if (!strv_isempty((context)->(name)) { \ 
                fprintf(f, "%s%s:", (prefix), (name)); \
                strv_fprintf(f, (context)->(name)); \
                fputs("\n", f); \
        } else

       STRV_PRINTF(c, f, prefix, noexec_paths, "NoExecPaths");
```

Reply:
no. please no macros. there's nothing wrong with functions.

Reply:
Fixed